File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.py
class Codec:
    code = "http://tinyurl.com/4e9iAk"
    def encode(self, longUrl: str) -> str:
        self.url = longUrl
        return Codec.code

    def decode(self, shortUrl: str) -> str:
        return self.url

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))

File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.cpp
class Solution {
public:
    string code = "http://tinyurl.com/4e9iAk";
    string url;
    // Encodes a URL to a shortened URL.
    string encode(string longUrl) {
        url = longUrl;
        return code;
    }

    // Decodes a shortened URL to its original URL.
    string decode(string shortUrl) {
        return url;
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));